src/main/java/org/terasology/namegenerator/town/TownAssetTheme.java

```diff
@@ -48,20 +57,55 @@
      * @param affixPrefab valid prefab with {@link TownNameAffixComponent}
      */
     TownAssetTheme(String namePrefab, String affixPrefab) {
-        this(Assets.getPrefab(namePrefab).get(), Assets.getPrefab(affixPrefab).get());
+        this(fetchNameList(namePrefab), fetchPrefixesList(affixPrefab), fetchPostfixesList(affixPrefab));
+    }
+
+    TownAssetTheme(List<String> namePrefab, List<String> prefixPrefab, List<String> postfixPrefab) {
+        names = namePrefab;
+        prefixes = prefixPrefab;
+        postfixes  = postfixPrefab;
     }
 
     /**
-     * @param namePrefab valid prefab with {@link NameGeneratorComponent}
-     * @param affixPrefab valid prefab with {link {@link TownNameAffixComponent}
+     * Fetch a name list from a String naming a {@code Prefab}.
+     *
+     * The named {@link Prefab} should contain a {@link NameGeneratorComponent}.
+     *
+     * @param s the name of a prefab, null returns an empty list
+     * @return the list of names associated with the named prefab
+     * @throws java.util.NoSuchElementException if {@code s} is not null but does not have an associated prefab
      */
-    TownAssetTheme(Prefab namePrefab, Prefab affixPrefab) {
-        NameGeneratorComponent basenames = namePrefab.getComponent(NameGeneratorComponent.class);
-        names = Collections.unmodifiableList(basenames.nameList);
+    private static List<String> fetchNameList(String s) {
+        if (s == null) {
```

Comment:
This may be simplified to something like
```java
private static List<String> fetchPrefixesList(String s) {    
    return 
        Assets.getPrefab(s)
            .map(prefab -> prefab.getComponent(TownNameAffixComponent.class))
            .map(comp -> comp.prefixes)
            .orElse(Collections.emptyList());
}
```